src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseArachneSecureServiceImpl.java

```diff
@@ -189,6 +189,15 @@ public BaseArachneSecureServiceImpl(UserStudyGroupedRepository userStudyGroupedR
         if (tenantRepository.findFirstByDataSourcesIdAndUsersId(dataSource.getId(), user.getId()).isPresent()) {
             participantRoles.add(ParticipantRole.DATA_SET_USER);
         }
+        final User standardUser = new User();
```

Comment:
Why do you duplicate code from "getRolesByDataNode", which is called a couple of lines above?